offset_calculation.py

```diff
@@ -1,38 +1,107 @@
-# 1 pedestal_file öffnen
-# plotten?
-# mittelwerte berechnen
+"""
+##########################################################
+* calculate offset constants and RMS from pedestal files *
+##########################################################
+
+annotations:
+
+- filedirectory is the path to the directory of your pedestal files
+- outputdirectory is the directory where the calibration constants are written to
+
+
+Usage:
+  offset_calculation.py <filedirectory> <outputdirectory>
+  offset_calculation.py (-h | --help)
+  offset_calculation.py --version
+
+Options:
+  -h --help     Show this screen.
+  --version     Show version.
+
+"""
 
 import numpy as np
 import matplotlib.pyplot as plt
-from dragonboard import EventGenerator
-
-def offset_calc(filename):
-	with open(filename, "rb") as f:
-		generator = EventGenerator(f, max_events=20)
-		#event = next(generator)
-		#plt.plot(event.data["low"][0])
-		#plt.show()
-		
-		# declare events which is a list of all generator events
-		events=list(generator)
-		
-		# read out list length
-		numevents=len(events)
-	
-		# initialize an array full of nans. length = number of caps, width = number of events
-		pedestalvalues = np.full((numevents, 4096),np.nan)
-		
-		#
-		pixelindex = 0
-		
-		for i in range(numevents):
-			stopcell = events[i].header.stop_cells["low"][pixelindex]
-			roi = events[i].roi
-			pedestalvalues[i][stopcell:stopcell+roi] = events[i].data["low"][pixelindex]
-			
-			plt.plot(pedestalvalues[i],".:")
-			
-		plt.show()
-		
+from dragonboard import EventGenerator # custom event generator
+from dragonboard.runningstats import RunningStats # Max's method to do statistics on the fly 
+from tqdm import tqdm # enable to show progress
+import glob # enable to search for files in directories
+import os
+from docopt import docopt # write your own menu! furthermore control terminal input parameters
+
+
+def offset_calc(filedirectory, outputdirectory, pixelindex, gaintype):
+    """ calculate mean offset & RMS for every capacitor and plot the data. the data is saved as .csv files.
+        The .csv files are named offsets_channel<pixelindex>_<gaintype>-gain. 
+        this FORMAT MUST NOT BE CHANGED
+    """
+
+    stats = RunningStats(shape=4096) # initialize stats array on which calculations are carried out
+
+    for filename in glob.glob(os.path.join(filedirectory, 'Ped*.dat')): # search for all .dat's in the current directory
+
+        with open(filename, "rb") as f: # open a file and call it f. "rb": read, file is binary
+
+            max_events = 1000 # 1000 at all
```

Comment:
Why `max_events = 1000`, in general, we want to use them all, right? Better just leave it out.